File: agent.py
import psutil
from fastapi import FastAPI

app = FastAPI()

@app.get("/status")
async def get_status():
    return {
        "cpu_usage": f"{psutil.cpu_percent(interval=1)}%",
        "ram_usage": f"{psutil.virtual_memory().percent}%",
        "disk_space": f"{psutil.disk_usage('/').used / (1024**3):.2f}GB / {psutil.disk_usage('/').total / (1024**3):.2f}GB"
    }

if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="0.0.0.0", port=8001)

File: agent.js
import express from 'express';
import si from 'systeminformation';

const app = express();
const port = 8001;

app.get('/status', async (req, res) => {
  try {
    const cpu = await si.currentLoad();
    const mem = await si.mem();
    const fs = await si.fsSize();

    let disk = fs.find(d => d.mount === '/');
    // If root disk not found, find the largest one
    if (!disk && fs.length > 0) {
        disk = fs.reduce((prev, curr) => (prev.size > curr.size) ? prev : curr);
    }

    res.json({
      cpu_usage: `${cpu.currentLoad.toFixed(2)}%`,
      ram_usage: `${((mem.active / mem.total) * 100).toFixed(2)}%`,
      disk_used: disk ? disk.used : 0, // in bytes
      disk_total: disk ? disk.size : 0, // in bytes
    });
  } catch (e) {
    console.error(e);
    res.status(500).send('Error getting system status');
  }
});

app.listen(port, '0.0.0.0', () => {
  console.log(`Agent listening at http://0.0.0.0:${port}`);
});
